### PR TITLE
Refactor Epilogue Screen

### LINKED ISSUE #14899: Android: Text on Blogging Reminders Epilogue screen should be updated to account for reminders removal
Internal ref: p1624247262227000-slack-C01UWB9GD1Q

### LINKED ISSUE #14912: Blogging Reminders: The reminder frequency and selected days should be formatted properly on the Epilogue screen
Internal ref: p1624247262227000-slack-C01UWB9GD1Q

### PR DESCRIPTION
**Pull request by ravishanker**
Updated Epilogue screen with a builder , latest copy and formatting

Fixes #14899 and #14912 

<table><tr>
<td></td>
<td></td>
<td></td>
</tr></table>

To test:

- Turn on the blogging reminders flag (Me/App settings/Test feature configuration - BloggingRemindersFeatureConfig)
- Notice the blogging goals option appears (My Site > Settings > Blogging goals)
- Click on for e.g. "Twice a week" button under Blogging goals setting
- Notice the blogging reminders prologue screen is shown
- Click on "Set goals"
- Select days screen is shown, select days, then "Notify me" button is enabled
- Click on "Notify me" button
- Notice the epilogue screen is shown. Done button is enabled, Clicking Done button dismisses bottomsheet
- Go back to Settings and de-select all days and see that epilogue shows a "no reminders set" message
- Go back to Settings and select all days and see that epilogue show "everyday" message

## Regression Notes
1. Potential unintended areas of impact
Site Settings

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Blogging Reminders flow

3. What automated tests I added (or what prevented me from doing so)
Unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt
Comment (planarvoid) [inline WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt]: Very minor: I think I'd move this directly to where it's used since it's not used in every case.

**Comment by planarvoid** [comment]
hey @ravishanker , thanks for the PR! I have a few minor observations which could be hard to implement: 
- For the "You'll get reminders to blog Four times a week..." - I think the number (in this case "Four") should be lower case
- I'm wondering if we have too many things bold in the message. I think only "Four" should be bold and from the listed days probably just the day names, not the commas and the "and". However, this might be quite hard so we can reconsider it.

**Comment by ravishanker** [comment]
> hey @ravishanker , thanks for the PR! I have a few minor observations which could be hard to implement:
> 
>     * For the "You'll get reminders to blog Four times a week..." - I think the number (in this case "Four") should be lower case
> 
>     * I'm wondering if we have too many things bold in the message. I think only "Four" should be bold and from the listed days probably just the day names, not the commas and the "and". However, this might be quite hard so we can reconsider it.

You're right, It was numbers like **2** times before.  After discussion and for consistency, chose to go with what you have done already Once, twice, three times ...   

Some bold formatting is harder, for now used sane defaults.  

Some formatting like lowercasing etc I'm waiting for desugaring PR to be merged.   Looks like that's happening soon given build pipeline just moved to BuildKite

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt
Comment (renanferrari) [inline WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt]: Since `enabledDays` is a set and a set doesn't guarantee the order of its elements, we should probably call `.sorted()` after mapping to `DayOfWeek` here, just to be safe.

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt
Comment (renanferrari) [inline WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt]: Is there any reason why we are using `.count()` here instead of `size`?

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt
Comment (renanferrari) [inline WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt]: Is this hardcoded `" and "` intentional? I suppose we should have a translatable string here instead at least?

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt
Comment (renanferrari) [inline WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt]: After discussing about this with the iOS folks, I believe that instead of trying to handle this logic ourselves, I'd recommend you to take a look at the [`ListFormatter`](https://developer.android.com/reference/android/icu/text/ListFormatter) class, which should make this way easier and much more accurate in terms of i18n.

**Review by renanferrari — COMMENTED** [review]
Hey @ravishanker 👋 Thanks for taking care of this!

>I'm waiting for desugaring PR to be merged

Ideally, the PR should only be ready for review when all of those dependencies have been resolved. In this particular case however, since the desugaring PR had already been merged to our feature branch (the one at #14864), this PR should be targeting that instead of `develop`. After all, the sole reason we have that feature branch is so that PRs like this wouldn't have to wait for those dependencies to be merged to `develop`.

Apart from that and from what @planarvoid already mentioned, I also left a few suggestions below.

File: WordPress/src/main/res/values/strings.xml
Comment (renanferrari) [inline WordPress/src/main/res/values/strings.xml]: Also, just to make sure you don't miss it, we have an updated spec for this string. Internal ref: pbArwn-2ox-p2

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt
Comment (ravishanker) [reply WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt]: No particular reason!  updated to use size now.  thanks

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt
Comment (ravishanker) [reply WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt]: all this should be replaced once formatting options are available after merging desugaring

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt
Comment (ravishanker) [reply WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt]: ListFormatter is only available from android O.  I was looking at using [Phrase](https://github.com/square/phrase) incase desugaring doesn't get merged or do something feasible for now

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt
Comment (ravishanker) [reply WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt]: I've refactored formatting as shown in the first screen shot above.  It is the closest to the spec that I could get.

**Comment by planarvoid** [comment]
Thanks for the changes @ravishanker , I've changed a few things to improve the text formatting so it looks like this now. Could you please review my changes @renanferrari ? 

Screenshot 2021-06-25 at 11 02 02

Screenshot 2021-06-25 at 11 02 15

**Review by renanferrari — APPROVED** [review]
@ravishanker @planarvoid Thanks for the changes!

Code is looking good and everything is working as expected.

LGTM 👍

**Pull request merged**
